From f54b22603dc46c33148c9d109d51f929fb59c174 Mon Sep 17 00:00:00 2001
From: Alex Cruz <du.alexcruz@gmail.com>
Date: Fri, 10 Nov 2017 20:41:55 -0500
Subject: [PATCH] Fix NPE when AOSP ambient display preference is removed

 Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 
 'void android.support.v7.preference.Preference.setOnPreferenceChangeListener(android.support.v7.preference.Preference$OnPreferenceChangeListener)'  on a null object reference
 at com.android.settingslib.core.AbstractPreferenceController.displayPreference(AbstractPreferenceController.java:27)
 at com.android.settings.dashboard.DashboardFragment.displayResourceTiles(DashboardFragment.java:239)
 at com.android.settings.dashboard.DashboardFragment.refreshAllPreferences(DashboardFragment.java:277)

diff --git a/packages/SettingsLib/src/com/android/settingslib/core/AbstractPreferenceController.java b/packages/SettingsLib/src/com/android/settingslib/core/AbstractPreferenceController.java
index e8d4c98..3904821 100644
--- a/packages/SettingsLib/src/com/android/settingslib/core/AbstractPreferenceController.java
+++ b/packages/SettingsLib/src/com/android/settingslib/core/AbstractPreferenceController.java
@@ -23,8 +23,10 @@ public void displayPreference(PreferenceScreen screen) {
       if (isAvailable()) {
           if (this instanceof Preference.OnPreferenceChangeListener) {
               final Preference preference = screen.findPreference(getPreferenceKey());
-              preference.setOnPreferenceChangeListener(
-                      (Preference.OnPreferenceChangeListener) this);
+                if (preference != null) {
+                    preference.setOnPreferenceChangeListener(
+                            (Preference.OnPreferenceChangeListener) this);
+                }
           }
       } else {
           removePreference(screen, getPreferenceKey());
